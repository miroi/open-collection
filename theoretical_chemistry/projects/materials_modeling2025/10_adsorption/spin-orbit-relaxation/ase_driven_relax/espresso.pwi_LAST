&CONTROL
   verbosity        = 'low'
   tstress          = .true.
   tprnfor          = .true.
   outdir           = './tmp'
   prefix           = 'c_hg'
   pseudo_dir       = './'
/
&SYSTEM
   ibrav            = 0
   nat              = 9
   ntyp             = 2
   ecutwfc          = 65
   occupations      = 'smearing'
   degauss          = 0.01
   smearing         = 'gauss'
   noncolin         = .true.
   lspinorb         = .true.
   assume_isolated  = '2D'
   vdw_corr         = 'DFT-D3'
   starting_magnetization(1) = 0.0
   starting_magnetization(2) = 0.0
/
&ELECTRONS
   conv_thr         = 1e-10
/
&IONS
/
&CELL
/
&FCP
/
&RISM
/
ATOMIC_SPECIES
C  12.011 C_r.upf
Hg  200.592 Hg-sp_r.upf

K_POINTS automatic
2 2 1  0 0 0

CELL_PARAMETERS angstrom
4.91200017930000 0.00000000000000 0.00000000000000
-2.45600065610000 4.25391661160000 0.00000000000000
0.00000000000000 0.00000000000000 15.00000000000000

ATOMIC_POSITIONS angstrom
C  0.0000000000 0.0001481856 7.4565810343  
C  -1.2278718566 2.1268841769 7.4565810343  
C  2.4560000900 0.0000000000 7.4633449543  
C  1.2278712906 2.1268841769 7.4565810343  
C  1.2279309585 0.7090260158 7.4605167432  
C  -0.0000003780 2.8358648471 7.4605167432  
C  3.6840688855 0.7090260158 7.4605167432  
C  2.4559997120 2.8359442390 7.4512167563  
Hg  2.4559997120 2.8359442390 10.8341452861  

