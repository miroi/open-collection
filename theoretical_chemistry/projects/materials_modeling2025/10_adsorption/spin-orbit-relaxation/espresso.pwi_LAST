&CONTROL
   verbosity        = 'low'
   tstress          = .true.
   tprnfor          = .true.
   outdir           = './tmp'
   prefix           = 'c_hg'
   pseudo_dir       = './'
/
&SYSTEM
   ibrav            = 0
   nat              = 9
   ntyp             = 2
   ecutwfc          = 65
   occupations      = 'smearing'
   degauss          = 0.01
   smearing         = 'gauss'
   noncolin         = .true.
   lspinorb         = .true.
   assume_isolated  = '2D'
   vdw_corr         = 'DFT-D3'
   starting_magnetization(1) = 0.0
   starting_magnetization(2) = 0.0
/
&ELECTRONS
   conv_thr         = 1e-10
/
&IONS
/
&CELL
/
&FCP
/
&RISM
/
ATOMIC_SPECIES
C  12.011 C_r.upf
Hg  200.592 Hg-sp_r.upf

K_POINTS automatic
2 2 1  0 0 0

CELL_PARAMETERS angstrom
4.91200017930000 0.00000000000000 0.00000000000000
-2.45600065610000 4.25391661160000 0.00000000000000
0.00000000000000 0.00000000000000 15.00000000000000

ATOMIC_POSITIONS angstrom
C  0.0000000000 0.0001490009 7.4566279210  
C  -1.2278711386 2.1268837714 7.4566279210  
C  2.4560000900 0.0000000000 7.4630059115  
C  1.2278705726 2.1268837714 7.4566279210  
C  1.2279301724 0.7090264656 7.4603351987  
C  -0.0000003780 2.8358639294 7.4603351987  
C  3.6840696716 0.7090264656 7.4603351987  
C  2.4559997120 2.8359442390 7.4517131370  
Hg  2.4559997120 2.8359442390 10.8343919637  

