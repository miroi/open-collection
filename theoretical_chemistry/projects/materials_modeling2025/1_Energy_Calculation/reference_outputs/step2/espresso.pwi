&CONTROL
   calculation      = 'scf'
   pseudo_dir       = './'
/
&SYSTEM
   ibrav            = 0
   nat              = 2
   ntyp             = 1
   ecutwfc          = 30
   occupations      = 'smearing'
   degauss          = 0.01
   smearing         = 'gauss'
/
&ELECTRONS
   conv_thr         = 1e-10
/
&IONS
/
&CELL
/
&FCP
/
&RISM
/
ATOMIC_SPECIES
Si 28.085 Si.upf

K_POINTS automatic
2 2 2  0 0 0

CELL_PARAMETERS angstrom
0.00000000000000 2.71490000000000 2.71490000000000
2.71490000000000 0.00000000000000 2.71490000000000
2.71490000000000 2.71490000000000 0.00000000000000

ATOMIC_POSITIONS angstrom
Si 1.3574500000 4.0723500000 4.0723500000  
Si 0.0000000000 0.0000000000 0.0000000000  

