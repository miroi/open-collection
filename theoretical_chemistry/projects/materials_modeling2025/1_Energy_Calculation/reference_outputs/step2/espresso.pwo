
     Program PWSCF v.7.4.1 starts on 27Aug2025 at  9:44: 1 

     This program is part of the open-source Quantum ESPRESSO suite
     for quantum simulation of materials; please cite
         "P. Giannozzi et al., J. Phys.:Condens. Matter 21 395502 (2009);
         "P. Giannozzi et al., J. Phys.:Condens. Matter 29 465901 (2017);
         "P. Giannozzi et al., J. Chem. Phys. 152 154105 (2020);
          URL http://www.quantum-espresso.org", 
     in publications or presentations arising from this work. More details at
     http://www.quantum-espresso.org/quote

     Parallel version (MPI), running on     4 processors

     MPI processes distributed on     1 nodes
     15051 MiB available memory on the printing compute node when the environment starts

     Reading input from espresso.pwi
Warning: card &CELL ignored
Warning: card / ignored
Warning: card &FCP ignored
Warning: card / ignored
Warning: card &RISM ignored
Warning: card / ignored

     Current dimensions of program PWSCF are:
     Max number of different atomic species (ntypx) = 10
     Max number of k-points (npk) =  40000
     Max angular momentum in pseudopotentials (lmaxx) =  4
     Message from routine setup:
     using ibrav=0 with symmetry is DISCOURAGED, use correct ibrav instead

     R & G space division:  proc/nbgrp/npool/nimage =       4
     Subspace diagonalization in iterative solution of the eigenvalue problem:
     a serial algorithm will be used


     Parallelization info
     --------------------
     sticks:   dense  smooth     PW     G-vecs:    dense   smooth      PW
     Min         108     108     31                 1495     1495     241
     Max         109     109     32                 1498     1498     242
     Sum         433     433    127                 5985     5985     965

     Using Slab Decomposition



     bravais-lattice index     =            0
     lattice parameter (alat)  =       7.2555  a.u.
     unit-cell volume          =     270.0773 (a.u.)^3
     number of atoms/cell      =            2
     number of atomic types    =            1
     number of electrons       =         8.00
     number of Kohn-Sham states=            8
     kinetic-energy cutoff     =      30.0000  Ry
     charge density cutoff     =     120.0000  Ry
     scf convergence threshold =      1.0E-10
     mixing beta               =       0.7000
     number of iterations used =            8  plain     mixing
     Exchange-correlation= PBE
                           (   1   4   3   4   0   0   0)

     celldm(1)=   7.255506  celldm(2)=   0.000000  celldm(3)=   0.000000
     celldm(4)=   0.000000  celldm(5)=   0.000000  celldm(6)=   0.000000

     crystal axes: (cart. coord. in units of alat)
               a(1) = (   0.000000   0.707107   0.707107 )  
               a(2) = (   0.707107   0.000000   0.707107 )  
               a(3) = (   0.707107   0.707107   0.000000 )  

     reciprocal axes: (cart. coord. in units 2 pi/alat)
               b(1) = ( -0.707107  0.707107  0.707107 )  
               b(2) = (  0.707107 -0.707107  0.707107 )  
               b(3) = (  0.707107  0.707107 -0.707107 )  


     PseudoPot. # 1 for Si read from file:
     ./Si.upf
     MD5 check sum: 02fab3f35e82123ef5bf1cb05d5b1a5e
     Pseudo is Norm-conserving + core correction, Zval =  4.0
     Generated using ONCVPSP code by D. R. Hamann
     Using radial grid of 1510 points,  6 beta functions with: 
                l(1) =   0
                l(2) =   0
                l(3) =   1
                l(4) =   1
                l(5) =   2
                l(6) =   2

     atomic species   valence    mass     pseudopotential
     Si                4.00    28.08500     Si( 1.00)

     48 Sym. Ops., with inversion, found (24 have fractional translation)



   Cartesian axes

     site n.     atom                  positions (alat units)
         1        Si     tau(   1) = (   0.3535534   1.0606602   1.0606602  )
         2        Si     tau(   2) = (   0.0000000   0.0000000   0.0000000  )

     number of k points=     3  Gaussian smearing, width (Ry)=  0.0100
                       cart. coord. in units 2pi/alat
        k(    1) = (   0.0000000   0.0000000   0.0000000), wk =   0.2500000
        k(    2) = (  -0.3535534  -0.3535534   0.3535534), wk =   1.0000000
        k(    3) = (  -0.7071068   0.0000000   0.0000000), wk =   0.7500000

     Dense  grid:     5985 G-vectors     FFT dimensions: (  32,  32,  32)

     Estimated max dynamical RAM per process >       2.37 MB

     Estimated total dynamical RAM >       9.47 MB

     Initial potential from superposition of free atoms

     starting charge       7.9988, renormalised to       8.0000
     Starting wfcs are    8 randomized atomic wfcs

     total cpu time spent up to now is        0.1 secs

     Self-consistent Calculation

     iteration #  1     ecut=    30.00 Ry     beta= 0.70
     Davidson diagonalization with overlap
     ethr =  1.00E-02,  avg # of iterations =  4.0

     total cpu time spent up to now is        0.1 secs

     total energy              =     -16.73075763 Ry
     estimated scf accuracy    <       0.09966993 Ry

     iteration #  2     ecut=    30.00 Ry     beta= 0.70
     Davidson diagonalization with overlap
     ethr =  1.25E-03,  avg # of iterations =  1.0

     total cpu time spent up to now is        0.1 secs

     total energy              =     -16.73591783 Ry
     estimated scf accuracy    <       0.00502618 Ry

     iteration #  3     ecut=    30.00 Ry     beta= 0.70
     Davidson diagonalization with overlap
     ethr =  6.28E-05,  avg # of iterations =  2.7

     total cpu time spent up to now is        0.1 secs

     total energy              =     -16.73645317 Ry
     estimated scf accuracy    <       0.00006714 Ry

     iteration #  4     ecut=    30.00 Ry     beta= 0.70
     Davidson diagonalization with overlap
     ethr =  8.39E-07,  avg # of iterations =  3.3

     total cpu time spent up to now is        0.1 secs

     total energy              =     -16.73649644 Ry
     estimated scf accuracy    <       0.00000083 Ry

     iteration #  5     ecut=    30.00 Ry     beta= 0.70
     Davidson diagonalization with overlap
     ethr =  1.04E-08,  avg # of iterations =  3.3

     total cpu time spent up to now is        0.1 secs

     total energy              =     -16.73649708 Ry
     estimated scf accuracy    <       0.00000006 Ry

     iteration #  6     ecut=    30.00 Ry     beta= 0.70
     Davidson diagonalization with overlap
     ethr =  6.90E-10,  avg # of iterations =  2.0

     total cpu time spent up to now is        0.1 secs

     total energy              =     -16.73649709 Ry
     estimated scf accuracy    <          1.4E-09 Ry

     iteration #  7     ecut=    30.00 Ry     beta= 0.70
     Davidson diagonalization with overlap
     ethr =  1.73E-11,  avg # of iterations =  4.0

     total cpu time spent up to now is        0.2 secs

     End of self-consistent calculation

          k = 0.0000 0.0000 0.0000 (   749 PWs)   bands (ev):

    -5.4900   6.5354   6.5354   6.5354   8.9700   8.9700   8.9700   9.9014

          k =-0.3536-0.3536 0.3536 (   754 PWs)   bands (ev):

    -3.1388  -0.5106   5.3035   5.3035   8.0212   9.7355   9.7355  14.1818

          k =-0.7071 0.0000 0.0000 (   740 PWs)   bands (ev):

    -1.3240  -1.3240   3.6123   3.6123   7.1310   7.1310  16.5743  16.5743

     the Fermi energy is     6.8233 ev

!    total energy              =     -16.73649709 Ry
     estimated scf accuracy    <          8.6E-12 Ry
     smearing contrib. (-TS)   =      -0.00004943 Ry
     internal energy E=F+TS    =     -16.73644766 Ry

     The total energy is F=E-TS. E is the sum of the following terms:
     one-electron contribution =       5.06240778 Ry
     hartree contribution      =       1.26277643 Ry
     xc contribution           =      -6.26206933 Ry
     ewald contribution        =     -16.79956254 Ry

     convergence has been achieved in   7 iterations

     Writing all to output data dir ./pwscf.save/ :
     XML data file, charge density, pseudopotentials, collected wavefunctions

     init_run     :      0.02s CPU      0.03s WALL (       1 calls)
     electrons    :      0.09s CPU      0.09s WALL (       1 calls)

     Called by init_run:
     wfcinit      :      0.00s CPU      0.01s WALL (       1 calls)
     potinit      :      0.00s CPU      0.01s WALL (       1 calls)
     hinit0       :      0.01s CPU      0.01s WALL (       1 calls)

     Called by electrons:
     c_bands      :      0.06s CPU      0.06s WALL (       7 calls)
     sum_band     :      0.01s CPU      0.01s WALL (       7 calls)
     v_of_rho     :      0.01s CPU      0.02s WALL (       8 calls)
     mix_rho      :      0.00s CPU      0.00s WALL (       7 calls)

     Called by c_bands:
     init_us_2    :      0.00s CPU      0.00s WALL (      45 calls)
     cegterg      :      0.06s CPU      0.06s WALL (      21 calls)

     Called by *egterg:
     cdiaghg      :      0.00s CPU      0.00s WALL (      82 calls)
     h_psi        :      0.05s CPU      0.05s WALL (      85 calls)
     g_psi        :      0.00s CPU      0.00s WALL (      61 calls)

     Called by h_psi:
     h_psi:calbec :      0.00s CPU      0.00s WALL (      85 calls)
     vloc_psi     :      0.05s CPU      0.05s WALL (      85 calls)
     add_vuspsi   :      0.00s CPU      0.00s WALL (      85 calls)

     General routines
     calbec       :      0.00s CPU      0.00s WALL (      85 calls)
     fft          :      0.01s CPU      0.01s WALL (      80 calls)
     ffts         :      0.00s CPU      0.00s WALL (       7 calls)
     fftw         :      0.05s CPU      0.05s WALL (    1146 calls)

     Parallel routines

     PWSCF        :      0.13s CPU      0.16s WALL


   This run was terminated on:   9:44: 1  27Aug2025            

=------------------------------------------------------------------------------=
   JOB DONE.
=------------------------------------------------------------------------------=
