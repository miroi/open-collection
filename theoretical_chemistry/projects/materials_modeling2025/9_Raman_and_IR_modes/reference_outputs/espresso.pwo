
     Program PWSCF v.7.4.1 starts on 27Aug2025 at  9:20:24 

     This program is part of the open-source Quantum ESPRESSO suite
     for quantum simulation of materials; please cite
         "P. Giannozzi et al., J. Phys.:Condens. Matter 21 395502 (2009);
         "P. Giannozzi et al., J. Phys.:Condens. Matter 29 465901 (2017);
         "P. Giannozzi et al., J. Chem. Phys. 152 154105 (2020);
          URL http://www.quantum-espresso.org", 
     in publications or presentations arising from this work. More details at
     http://www.quantum-espresso.org/quote

     Serial version
     14952 MiB available memory on the printing compute node when the environment starts

     Reading input from espresso.pwi
Warning: card &CELL ignored
Warning: card / ignored
Warning: card &FCP ignored
Warning: card / ignored
Warning: card &RISM ignored
Warning: card / ignored

     Current dimensions of program PWSCF are:
     Max number of different atomic species (ntypx) = 10
     Max number of k-points (npk) =  40000
     Max angular momentum in pseudopotentials (lmaxx) =  4
     Message from routine setup:
     no reason to have ecutrho>4*ecutwfc
     Message from routine setup:
     using ibrav=0 with symmetry is DISCOURAGED, use correct ibrav instead


     G-vector sticks info
     --------------------
     sticks:   dense  smooth     PW     G-vecs:    dense   smooth      PW
     Sum         877     439    139                17261     6183    1067

     Using Slab Decomposition



     bravais-lattice index     =            0
     lattice parameter (alat)  =       7.3065  a.u.
     unit-cell volume          =     275.8081 (a.u.)^3
     number of atoms/cell      =            2
     number of atomic types    =            1
     number of electrons       =         8.00
     number of Kohn-Sham states=            4
     kinetic-energy cutoff     =      30.0000  Ry
     charge density cutoff     =     240.0000  Ry
     scf convergence threshold =      1.0E-08
     mixing beta               =       0.7000
     number of iterations used =            8  plain     mixing
     Exchange-correlation=  SLA  PZ   NOGX NOGC
                           (   1   1   0   0   0   0   0)

     celldm(1)=   7.306465  celldm(2)=   0.000000  celldm(3)=   0.000000
     celldm(4)=   0.000000  celldm(5)=   0.000000  celldm(6)=   0.000000

     crystal axes: (cart. coord. in units of alat)
               a(1) = (  -0.000000   0.707107   0.707107 )  
               a(2) = (   0.707107   0.000000   0.707107 )  
               a(3) = (   0.707107   0.707107   0.000000 )  

     reciprocal axes: (cart. coord. in units 2 pi/alat)
               b(1) = ( -0.707107  0.707107  0.707107 )  
               b(2) = (  0.707107 -0.707107  0.707107 )  
               b(3) = (  0.707107  0.707107 -0.707107 )  


     PseudoPot. # 1 for Si read from file:
     ./Si.pz-vbc.UPF
     MD5 check sum: 6dfa03ddd5817404712e03e4d12deb78
     Pseudo is Norm-conserving, Zval =  4.0
     Generated by new atomic code, or converted to UPF format
     Using radial grid of  431 points,  2 beta functions with: 
                l(1) =   0
                l(2) =   1

     atomic species   valence    mass     pseudopotential
     Si                4.00    28.08500     Si( 1.00)

     48 Sym. Ops., with inversion, found (24 have fractional translation)



   Cartesian axes

     site n.     atom                  positions (alat units)
         1        Si     tau(   1) = (   0.3535534   1.0606602   1.0606602  )
         2        Si     tau(   2) = (   0.0000000   0.0000000   0.0000000  )

     number of k points=     3
                       cart. coord. in units 2pi/alat
        k(    1) = (   0.0000000   0.0000000   0.0000000), wk =   0.2500000
        k(    2) = (  -0.3535534  -0.3535534   0.3535534), wk =   1.0000000
        k(    3) = (  -0.7071068   0.0000000   0.0000000), wk =   0.7500000

     Dense  grid:    17261 G-vectors     FFT dimensions: (  40,  40,  40)

     Smooth grid:     6183 G-vectors     FFT dimensions: (  32,  32,  32)

     Estimated max dynamical RAM per process >       7.71 MB

     Initial potential from superposition of free atoms

     starting charge       7.9990, renormalised to       8.0000
     Starting wfcs are    8 randomized atomic wfcs

     total cpu time spent up to now is        0.1 secs

     Self-consistent Calculation

     iteration #  1     ecut=    30.00 Ry     beta= 0.70
     Davidson diagonalization with overlap
     ethr =  1.00E-02,  avg # of iterations =  2.0

     total cpu time spent up to now is        0.1 secs

     total energy              =     -15.65850333 Ry
     estimated scf accuracy    <       0.09927018 Ry

     iteration #  2     ecut=    30.00 Ry     beta= 0.70
     Davidson diagonalization with overlap
     ethr =  1.24E-03,  avg # of iterations =  1.3

     total cpu time spent up to now is        0.1 secs

     total energy              =     -15.66368231 Ry
     estimated scf accuracy    <       0.00500125 Ry

     iteration #  3     ecut=    30.00 Ry     beta= 0.70
     Davidson diagonalization with overlap
     ethr =  6.25E-05,  avg # of iterations =  2.0

     total cpu time spent up to now is        0.2 secs

     total energy              =     -15.66428048 Ry
     estimated scf accuracy    <       0.00008195 Ry

     iteration #  4     ecut=    30.00 Ry     beta= 0.70
     Davidson diagonalization with overlap
     ethr =  1.02E-06,  avg # of iterations =  2.7

     total cpu time spent up to now is        0.2 secs

     total energy              =     -15.66432273 Ry
     estimated scf accuracy    <       0.00000199 Ry

     iteration #  5     ecut=    30.00 Ry     beta= 0.70
     Davidson diagonalization with overlap
     ethr =  2.49E-08,  avg # of iterations =  4.7

     total cpu time spent up to now is        0.2 secs

     total energy              =     -15.66432441 Ry
     estimated scf accuracy    <       0.00000012 Ry

     iteration #  6     ecut=    30.00 Ry     beta= 0.70
     Davidson diagonalization with overlap
     ethr =  1.50E-09,  avg # of iterations =  4.0

     total cpu time spent up to now is        0.2 secs

     End of self-consistent calculation

          k = 0.0000 0.0000 0.0000 (   749 PWs)   bands (ev):

    -5.7436   6.1209   6.1209   6.1209

          k =-0.3536-0.3536 0.3536 (   778 PWs)   bands (ev):

    -3.4170  -0.8567   4.9114   4.9114

          k =-0.7071 0.0000 0.0000 (   774 PWs)   bands (ev):

    -1.6371  -1.6371   3.2438   3.2438

     highest occupied level (ev):     6.1209

!    total energy              =     -15.66432448 Ry
     estimated scf accuracy    <          8.3E-10 Ry

     The total energy is the sum of the following terms:
     one-electron contribution =       4.58857210 Ry
     hartree contribution      =       1.27069648 Ry
     xc contribution           =      -4.84119955 Ry
     ewald contribution        =     -16.68239351 Ry

     convergence has been achieved in   6 iterations

     Forces acting on atoms (cartesian axes, Ry/au):

     atom    1 type  1   force =    -0.00000000    0.00000000    0.00000000
     atom    2 type  1   force =    -0.00000000   -0.00000000    0.00000000

     Total force =     0.000000     Total SCF correction =     0.000000


     Computing stress (Cartesian axis) and pressure

          total   stress  (Ry/bohr**3)                   (kbar)     P=       35.49
   0.00024127   0.00000000   0.00000000           35.49        0.00        0.00
   0.00000000   0.00024127   0.00000000            0.00       35.49        0.00
   0.00000000   0.00000000   0.00024127            0.00        0.00       35.49


     Writing all to output data dir ./tmp/si_scf.save/ :
     XML data file, charge density, pseudopotentials, collected wavefunctions

     init_run     :      0.04s CPU      0.05s WALL (       1 calls)
     electrons    :      0.16s CPU      0.16s WALL (       1 calls)
     forces       :      0.00s CPU      0.00s WALL (       1 calls)
     stress       :      0.01s CPU      0.02s WALL (       1 calls)

     Called by init_run:
     wfcinit      :      0.01s CPU      0.01s WALL (       1 calls)
     potinit      :      0.01s CPU      0.01s WALL (       1 calls)
     hinit0       :      0.01s CPU      0.01s WALL (       1 calls)

     Called by electrons:
     c_bands      :      0.09s CPU      0.09s WALL (       6 calls)
     sum_band     :      0.03s CPU      0.03s WALL (       6 calls)
     v_of_rho     :      0.02s CPU      0.02s WALL (       7 calls)
     mix_rho      :      0.01s CPU      0.01s WALL (       6 calls)

     Called by c_bands:
     init_us_2    :      0.00s CPU      0.00s WALL (      45 calls)
     cegterg      :      0.09s CPU      0.09s WALL (      18 calls)

     Called by *egterg:
     cdiaghg      :      0.00s CPU      0.00s WALL (      68 calls)
     h_psi        :      0.09s CPU      0.10s WALL (      71 calls)
     g_psi        :      0.00s CPU      0.00s WALL (      50 calls)

     Called by h_psi:
     h_psi:calbec :      0.00s CPU      0.00s WALL (      71 calls)
     vloc_psi     :      0.09s CPU      0.09s WALL (      71 calls)
     add_vuspsi   :      0.00s CPU      0.00s WALL (      71 calls)

     General routines
     calbec       :      0.00s CPU      0.00s WALL (     122 calls)
     fft          :      0.03s CPU      0.02s WALL (      37 calls)
     ffts         :      0.00s CPU      0.00s WALL (      13 calls)
     fftw         :      0.09s CPU      0.09s WALL (     580 calls)
     interpolate  :      0.01s CPU      0.01s WALL (       7 calls)
     davcio       :      0.00s CPU      0.00s WALL (     104 calls)


     PWSCF        :      0.24s CPU      0.27s WALL


   This run was terminated on:   9:20:24  27Aug2025            

=------------------------------------------------------------------------------=
   JOB DONE.
=------------------------------------------------------------------------------=
