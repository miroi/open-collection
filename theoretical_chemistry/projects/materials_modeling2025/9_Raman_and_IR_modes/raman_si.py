#!/usr/bin/env python
from ase import Atoms
from ase.calculators.espresso import Espresso, EspressoProfile
import os
import sys
import subprocess
import numpy as np
import matplotlib.pyplot as plt
import shutil
sys.stdout.flush() 

# ==============================================
# ASE DFPT Raman calculation
# Note: LDA + Norm-conserving psp needed for Raman
# ==============================================

# ==============================================
# 1. Quantum Espresso input parameters
# ==============================================
pseudopotentials = {'Si': 'Si.pz-vbc.UPF'} 

base_input_data = {
    'control': {
        'prefix': 'si_scf',
        'outdir': './tmp',
        'verbosity': 'low',
        'tstress': True,
        'tprnfor': True,
        'disk_io': 'high'
    },
    'system': {
        'ecutwfc': 30,
        'ecutrho': 240,
        'occupations': 'fixed',
        'ibrav': 0,
        'nat': 2,
        'ntyp': 1
    },
    'electrons': {
        'conv_thr': 1.0e-8
    }
}

# ==============================================
# 2. Atomic Structure 
# ==============================================
atoms = Atoms(
    symbols=['Si']*2,
    positions=[ # positions in Angstrom
        [1.3669840665567621, 4.1009521996702860, 4.1009521996702860],
        [0.0000000000000000, 0.0000000000000000, 0.0000000000000000]
    ],
    cell=[ # positions in Angstrom
        [-0.0000000000000000, 2.7339681331135242, 2.7339681331135242],
        [2.7339681331135242, 0.0000000000000000, 2.7339681331135242],
        [2.7339681331135242, 2.7339681331135242, 0.0000000000000000]
    ],
    pbc=[True, True, True]
)

# ==============================================
# 3. Calculator configuration
# ==============================================
# Set QE bin directory 
qe_bin = "/home/dsen/work/bin/qe-7.4.1_serial"
#qe_bin = "/home/dsen/work/bin/qe-7.4.1"

# Parallel calculation 
pw_command = f'{qe_bin}/bin/pw.x'
#pw_command = f'mpirun -np 4 {qe_bin}/bin/pw.x'
ph_command = f'{qe_bin}/bin/ph.x < ph.in > ph.out 2>&1'
#ph_command = f'mpirun -np 4 {qe_bin}/bin/ph.x < ph.in > ph.out 2>&1'

# Serial post-processing commands for fast execution
dynmat_command = f'{qe_bin}/bin/dynmat.x < dynmat.in > dynmat.out 2>&1'

pw_profile = EspressoProfile(
    command=pw_command,
    pseudo_dir='./'
)

# Set k-grids
scf_kpts = (2, 2, 2)

# QE tool 
def run_qe_tool(command, input_file, tool_name):
    """Run QE tool with input file and redirect output to file"""
    try:
        with open(input_file, 'r') as fin:
            subprocess.run(command, shell=True, stdin=fin, check=True)
        print(f"  {tool_name} completed successfully", flush=True)
        return True
    except subprocess.CalledProcessError as e:
        print(f"Error running {tool_name}: {e}", flush=True)
        return False

# ==============================================
# 4. SCF calculation
# ==============================================
scf_input_data = base_input_data.copy()
scf_input_data['control']['calculation'] = 'scf'

scf_calc = Espresso(
    profile=pw_profile,
    pseudopotentials=pseudopotentials,
    input_data=scf_input_data,
    kpts=scf_kpts
)

print("\nRunning SCF calculation...", flush=True)
os.makedirs('tmp', exist_ok=True)
atoms.calc = scf_calc
total_energy = atoms.get_potential_energy()
print(f"  Total energy: {total_energy:.6f} eV", flush=True)

# ==============================================
# 5. DFPT phonon calculation (Gamma only)
# ==============================================
print("\nRunning DFPT phonon calculation...", flush=True)

with open('ph.in', 'w') as f:
    f.write(f"""&INPUTPH
   prefix = '{base_input_data['control']['prefix']}',
   outdir = '{base_input_data['control']['outdir']}',
   tr2_ph = 1.0e-12,
   amass(1) = 28.0855,
   epsil = .true.,
   trans = .true.,
   lraman = .true.,
   fildyn = 'si.dyn',
   ldisp = .false.
/
0.0 0.0 0.0
""")

run_qe_tool(ph_command, 'ph.in', 'ph.x')

# ==============================================
# 6. Raman Calculation 
# ==============================================
print("\nCalculating Raman spectra...", flush=True) 

with open('dynmat.in', 'w') as f:
    f.write("""&INPUT
   fildyn = 'si.dyn',
   asr = 'crystal',
/
""")

run_qe_tool(dynmat_command, 'dynmat.in', 'dynmat.x')

# ==============================================
# 7. Clean up and organize files
# ==============================================
print("\nCleaning up and organizing files...", flush=True)
os.makedirs('vibrational_results', exist_ok=True)

# Move all calculation files to results directory
for fname in os.listdir('.'):
    if (fname in ['ph.in', 'ph.out', 'espresso.pwi', 'espresso.pwo', 'espresso.err'] or
        fname.endswith(('.dyn', '.in', '.mold', '.axsf')) or
        fname.startswith(('drho_e'))):
        shutil.move(fname, os.path.join('vibrational_results', fname))

print("  Moved all vibrational calculation files to vibrational_results directory", flush=True)

print("\n=== Raman Calculation Complete ===", flush=True)
print("Generated files:", flush=True)
print("- SCF calculation files: tmp/", flush=True)
print("- Phonon calculation files: vibrational_results/ph.*", flush=True)
print("- Dynamical matrix files: vibrational_results/si.dyn*", flush=True)
print("- Final Raman and IR data: dynmat.out", flush=True)