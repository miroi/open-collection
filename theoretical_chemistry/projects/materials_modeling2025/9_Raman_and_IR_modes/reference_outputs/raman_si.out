
Running SCF calculation...
  Total energy: -213.123973 eV

Running DFPT phonon calculation...
  ph.x completed successfully

Calculating Raman spectra...
  dynmat.x completed successfully

Cleaning up and organizing files...
  Moved all vibrational calculation files to vibrational_results directory

=== Raman Calculation Complete ===
Generated files:
- SCF calculation files: tmp/
- Phonon calculation files: vibrational_results/ph.*
- Dynamical matrix files: vibrational_results/si.dyn*
- Final Raman and IR data: dynmat.out
